File: src/types/assessment.ts
export interface AssessmentRequest {
  exerciseId: string;
  expectedText: string;
  userTranscript: string;
  language: string;
  childId?: string;
}

export type NextDifficulty = 'easy' | 'medium' | 'hard';

export interface AssessmentResponse {
  score: number;
  accuracy: number;
  fluency: number;
  skill: string;
  needsPractice: boolean;
  feedback: string;
  nextDifficulty: NextDifficulty;
}

File: app/schemas/assessment.py
from enum import Enum
from pydantic import BaseModel, Field

class Language(str, Enum):
    en = "en"
    ta = "ta"

class Skill(str, Enum):
    reading_accuracy = "reading_accuracy"
    reading_fluency = "reading_fluency"
    spelling = "spelling"
    sentence_formation = "sentence_formation"
    vocabulary = "vocabulary"

class Difficulty(str, Enum):
    easy = "easy"
    medium = "medium"
    hard = "hard"

class AssessmentRequest(BaseModel):
    exerciseId: str = Field(min_length=1)
    expectedText: str = Field(min_length=1)
    userTranscript: str = Field(min_length=1)
    language: Language

class AssessmentResponse(BaseModel):
    score: int = Field(ge=0, le=100)
    accuracy: int = Field(ge=0, le=100)
    fluency: int = Field(ge=0, le=100)
    skill: Skill
    needsPractice: bool
    feedback: str = Field(min_length=1)
    nextDifficulty: Difficulty
